### ISSUE TITLE
[Feature Request] Partial exhaustion indicator in UI

### ISSUE THREAD

**Issue by starslab** [munt/munt#103]
It would be nice to have some kind of indicator that notes are being dropped or not played due to partial exhaustion, especially because this is an emulator that can just have more partials, if the end-user knows it needs more. 😄 

It looks like there's a bunch of free room in the 'Synth' box in the UI, though it might make more sense to put it somewhere in the 'Partial State' area - maybe next to the 'Partial State' text itself?

**Comment by sergm** [comment]
Yes, this is an interesting idea, but unfortunately, it is not straightforward to implement. We do keep track of partial abortion already, albeit partials may be aborted even when you have plenty of free ones, depending on the assign mode. I'll see how best to check for partial exhaustion state yet avoid degrading performance significantly...

**Comment by sergm** [comment]
Perhaps, a good idea may be to show a four-state icon nearby the MIDI MESSAGE indicator (or even replacing it) with the same set of colors used for partial state indicator. For example, it may show:

- Grey - inactive synth state.
- Green - all used partials fit into the available partials, none is terminated prematurely.
- Yellow - some notes have been prematurely silenced due to a lack of free partials (shown for some time, e.g. 80 millis).
- Red - some notes were recently denied to play by the partial allocator due to insufficient of free partials.

I am on the fence whether this should be a separate indicator, or we could extend the semantic of MIDI MESSAGE optionally. Of course, there is a plenty of free space in the Synth box, but two sibling indicators seems a bit overkill. OTOH, the separate insufficient partial indicator could be made even more visible.

**Comment by starslab** [comment]
I'm certainly a fan of the existing MIDI MESSAGE display, as it mimics the real hardware display. It's unfortunate that Roland put it right smack dab in the middle of the MIDI MESSAGE text, as that leads to a lot of wasted space.

If you want to make it an option, perhaps.....
* Option off, existing behavior
* Option on, move the green flasher over to the side, and put your partial state indicator over to the other side, like this:

-------------------
| MIDI MESSAGE |
|     --      [    ]      |
-------------------

Maybe it's worth considering putting this indicator down next to the Partial State text instead - there's lots of room for it there, and it's consistent with the purpose of that display.

**Comment by sergm** [comment]
I suppose, the new indicator should be shown even when the lower synth details pane is closed. I'm going to place it close to the MIDI MESSAGE LED.
